src/index/vocabulary/PolymorphicVocabulary.h

```diff
@@ -87,6 +88,39 @@ class PolymorphicVocabulary {
         },
         vocab_);
   }
+
+  // Analogous to `lower_bound`, but since `word` is guaranteed to be a full
+  // word, not a prefix, this function can respect the split of an underlying
+  // `SplitVocabulary`.
+  template <typename String, typename Comp, bool getUpperBound>
```

Comment:
If you move the bool template parameter to the front, the compiler might be able to deduce the other two args automatically and you don't have to explicitly specify them down below.